### ISSUE TITLE
Global variables as PVs

### ISSUE THREAD

**Issue by georgekontogiorgos** [dls-controls/pmac#80]
Hello!

I would like to know if there is a way to read global variables on IOC level, as it does with M-variables on GPIO or `Plc[i].Running`. If there is any feature to read P-variables I guess that it could be possible.

On my project I will use the PV `SendCmd` to read custom variables but I would like to know if there is a fashionable form to do it.

Thanks
George

**Comment by gilesknap** [comment]
Hi George,

Take a look at the template pmacApp/Db/pmacVariableRead.template

I think that will have what you need.

**Comment by georgekontogiorgos** [comment]
> Hi George,
> 
> Take a look at the template pmacApp/Db/pmacVariableRead.template
> 
> I think that will have what you need.
 
Hi Giles! 

Yes, it is exactly what I was searching for. But instead using `pmacApp/Db/pmacVariableRead.template` I used `pmacApp/Db/pmacVariableWrite.template`. I do not know why but the template you sent me throwed this error during IOC startup:

```
PV: MNC:EmergencyError:RBV scanAdd: I/O Intr not valid (no get_ioint_info)
```

Furthermore, caget and caput could not catch the correct global variable. So I tried `pmacVariableWrite.template` and it worked fine for me, besides it, this template already have `RBV` and `SET` "fields".

Thanks a lot for replying and for helping me.
